app.py

```diff
@@ -673,10 +665,12 @@ def edit():
 		session['actions'][DECTREE] += int(request.args.get('dec', 0))
 	session.modified = True
 	msg = ''
-	if request.args.get('annotated', False):
+	if request.args.get('annotated') == '1':
 		msg = Markup('<font color=red>You have already annotated '
 				'this sentence.</font><button id="undo" onclick="undoAccept()">Delete tree from database</button>')
-		treeobj = ActivedopTree.from_str(request.args.get('tree'))
+		id = QUEUE[sentno - 1][3]
+		treestr, n = getannotation(username, id)
+		treeobj = ActivedopTree.from_str(treestr)
```

Comment:
`annotated=1` means the caller says an annotation for the sentence is saved in the database, correct? And `getannotation()` is querying the database for that annotation?

Reply:
Suppose we are annotating a fresh sentence, have selected a parse from the parser and clicked "edit". Does that tree get sent in the request? Is it now POST rather than GET?

Reply:
> `annotated=1` means the caller says an annotation for the sentence is saved in the database, correct? And `getannotation()` is querying the database for that annotation?

Yes, correct on both counts!

> Suppose we are annotating a fresh sentence, have selected a parse from the parser and clicked "edit". Does that tree get sent in the request? Is it now POST rather than GET?

No, the tree does not get sent in that request (as before) and it's a GET request (as before). In this case, the URL parameter `n` is passed in the GET request, and `n` is used to identify the parse tree the user wishes to edit: see https://github.com/nschneid/activedop/blob/bd6ead7ab406a80aed3e2ce398008240cf2250e5/app.py#L674

Reply:
OK. When does this bit get invoked?
https://github.com/nschneid/activedop/blob/bd6ead7ab406a80aed3e2ce398008240cf2250e5/app.py#L687-L689

Reply:
Ah, that's a vestige of the old logic. Removed in https://github.com/nschneid/activedop/pull/94/commits/26a21e6e9133dbce2ed2f830f2583532d743e8be